vcd/catalogitem.go

```diff
@@ -55,18 +56,30 @@ func findCatalogItem(d *schema.ResourceData, vcdClient *VCDClient) (*govcd.Catal
 
 	catalog, err := adminOrg.GetCatalogByName(d.Get("catalog").(string), false)
 	if err != nil {
-		log.Printf("[DEBUG] Unable to find catalog. Removing from tfstate")
-		d.SetId("")
+		log.Printf("[DEBUG] Unable to find catalog.")
 		return nil, fmt.Errorf("unable to find catalog: %s", err)
 	}
 
-	catalogItem, err := catalog.GetCatalogItemByName(d.Get("name").(string), false)
+	identifier := d.Id()
+
+	// check if identifier still is in deprecated style `catalogName:mediaName`
+	if identifier == "" || strings.Contains(identifier, ":") {
```

Comment:
Is this needed for a smooth upgrade from 2.4.0 to 2.5.0?

Reply:
Yes, cause in state file we could get old style id

Reply:
Got it. Please change/extend the comment with something like this:
```
// Check if identifier is still in deprecated style `catalogName:mediaName`
// Required for backwards compatibility as identifier has been changed to vCD ID in 2.5.0
```

Reply:
done